src/log.rs

```diff
@@ -27,3 +46,136 @@ macro_rules! ngx_log_debug_http {
         $crate::ngx_log_debug!(log, $($arg)*);
     }
 }
+
+/// Debug masks for use with ngx_log_debug_mask, these represent the only accepted values for the
+/// mask.
+#[derive(Debug)]
+pub enum DebugMasks {
+    /// Aligns to the NGX_LOG_DEBUG_CORE mask.
+    Core,
+    /// Aligns to the NGX_LOG_DEBUG_ALLOC mask.
+    Alloc,
+    /// Aligns to the NGX_LOG_DEBUG_MUTEX mask.
+    Mutex,
+    /// Aligns to the NGX_LOG_DEBUG_EVENT mask.
+    Event,
+    /// Aligns to the NGX_LOG_DEBUG_HTTP mask.
+    Http,
+    /// Aligns to the NGX_LOG_DEBUG_MAIL mask.
+    Mail,
+    /// Aligns to the NGX_LOG_DEBUG_STREAM mask.
+    Stream,
+}
+
+impl TryFrom<u32> for DebugMasks {
+    type Error = u32;
+
+    fn try_from(value: u32) -> Result<Self, Self::Error> {
+        match value {
+            crate::ffi::NGX_LOG_DEBUG_CORE => Ok(DebugMasks::Core),
+            crate::ffi::NGX_LOG_DEBUG_ALLOC => Ok(DebugMasks::Alloc),
+            crate::ffi::NGX_LOG_DEBUG_MUTEX => Ok(DebugMasks::Mutex),
+            crate::ffi::NGX_LOG_DEBUG_EVENT => Ok(DebugMasks::Event),
+            crate::ffi::NGX_LOG_DEBUG_HTTP => Ok(DebugMasks::Http),
+            crate::ffi::NGX_LOG_DEBUG_MAIL => Ok(DebugMasks::Mail),
+            crate::ffi::NGX_LOG_DEBUG_STREAM => Ok(DebugMasks::Stream),
+            _ => Err(0),
+        }
+    }
+}
+
+impl From<DebugMasks> for u32 {
+    fn from(value: DebugMasks) -> Self {
+        match value {
+            DebugMasks::Core => crate::ffi::NGX_LOG_DEBUG_CORE,
+            DebugMasks::Alloc => crate::ffi::NGX_LOG_DEBUG_ALLOC,
+            DebugMasks::Mutex => crate::ffi::NGX_LOG_DEBUG_MUTEX,
+            DebugMasks::Event => crate::ffi::NGX_LOG_DEBUG_EVENT,
+            DebugMasks::Http => crate::ffi::NGX_LOG_DEBUG_HTTP,
+            DebugMasks::Mail => crate::ffi::NGX_LOG_DEBUG_MAIL,
+            DebugMasks::Stream => crate::ffi::NGX_LOG_DEBUG_STREAM,
+        }
+    }
+}
+
+/// Log with appropriate debug mask.
+///
+/// When the request logger is available `ngx_log_debug_http` can be used for `NGX_LOG_DEBUG_HTTP` masks.
```

Comment:
seems to me the copy/paste was left as is... 

```suggestion
/// When the request logger is available `ngx_log_debug_mask` can be used for `NGX_LOG_DEBUG_HTTP` masks.
```

Reply:
This might be poor wording. This is correctly referring to ngx_log_debug_http. Its intent is to say ngx_log_debug_http should be preferred when you can use a Request.

Basically justifying ngx_log_debug_http AND ngx_log_debug_mask (for when you don't have a specific Request to use).